editor/src/core/shared/array-utils.ts

```diff
@@ -525,3 +525,25 @@ export function isPrefixOf<T>(
 export function valueOrArrayToArray<T>(ts: T | T[]): T[] {
   return Array.isArray(ts) ? ts : [ts]
 }
+
+export function createArrayWithLength<T>(length: number, value: T | ((index: number) => T)): T[] {
```

Comment:
I'd keep the function signature simple and only accept functions (this is the same with the builtin `map` function)